diff --git a/node_modules/@astrojs/starlight/schema.ts b/node_modules/@astrojs/starlight/schema.ts
index a90220b..8731349 100644
--- a/node_modules/@astrojs/starlight/schema.ts
+++ b/node_modules/@astrojs/starlight/schema.ts
@@ -33,7 +33,15 @@ const StarlightFrontmatterSchema = (context: SchemaContext) =>
 		head: HeadConfigSchema({ source: 'content' }),
 
 		/** Override global table of contents configuration for this page. */
-		tableOfContents: TableOfContentsSchema().optional(),
+		tableOfContents: z
+			.union([
+				z.object({
+					minHeadingLevel: z.int().min(1).max(6).optional(),
+					maxHeadingLevel: z.int().min(1).max(6).optional(),
+				}),
+				z.boolean(),
+			])
+			.optional(),
 
 		/**
 		 * Set the layout style for this page.
diff --git a/node_modules/@astrojs/starlight/utils/routing/data.ts b/node_modules/@astrojs/starlight/utils/routing/data.ts
index 25fdc6a..e313337 100644
--- a/node_modules/@astrojs/starlight/utils/routing/data.ts
+++ b/node_modules/@astrojs/starlight/utils/routing/data.ts
@@ -61,12 +61,20 @@ export function generateRouteData({
 }
 
 export function getToC({ entry, lang, headings }: PageProps) {
-	const tocConfig =
-		entry.data.template === 'splash'
-			? false
-			: entry.data.tableOfContents !== undefined
-				? entry.data.tableOfContents
-				: config.tableOfContents;
+	const frontmatterToC = entry.data.tableOfContents;
+	let tocConfig;
+	if (entry.data.template === 'splash') {
+		tocConfig = false;
+	} else if (frontmatterToC === undefined) {
+		tocConfig = config.tableOfContents;
+	} else if (typeof frontmatterToC === 'boolean') {
+		tocConfig = frontmatterToC ? config.tableOfContents : false;
+	} else {
+		tocConfig = {
+			minHeadingLevel: frontmatterToC.minHeadingLevel ?? config.tableOfContents.minHeadingLevel,
+			maxHeadingLevel: frontmatterToC.maxHeadingLevel ?? config.tableOfContents.maxHeadingLevel,
+		};
+	}
 	if (!tocConfig) return;
 	const t = useTranslations(lang);
 	return {
